gdal/frmts/tiledb/tiledbdataset.cpp

```diff
@@ -331,9 +333,13 @@ CPLErr TileDBRasterBand::IWriteBlock( int nBlockXOff,
             int nBytes = GDALGetDataTypeSizeBytes( eAttrType );
             int nValues = nBlockXSize * nBlockYSize;
             void* pAttrBlock = VSIMalloc( nBytes * nValues );
+            aBlocks.push_back( pAttrBlock );
```

Comment:
I'd make use of RAII

1. `std::vector<std::unique_ptr<void, decltype(&VSIFree)>> aBlocks;`
2. aBlocks.emplace_back(pAttrBlock, &VSIFree);

and remove all the `for ( auto pBlk: aBlocks )` as no longer necessary.

Quick comment w/o checking the wider scope, so may be invalid.

Reply:
@mloskot thanks, I will make that change